### PR TITLE
Replace blacklisting term

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

> **Note**: Right now we use blacklisting as a term when a process instance is marked as there is no continuation is allowed or possible. In essence, we ban the instance from further progressing. 

This concept is unfortunately already a long time overdue (to be replaced) but we have had not enough time for it yet. Due to our growing community, it makes sense to document also such edge-case aspects of our system, since more and more people running into this.

In order to make the world a better place and to document and speak about this concept in public it also makes sense to not use such a racially overloaded term.

With this PR I tried to replace the term with a less bad name like: ban. In consequence, we need to adjust other places where we use the old term ofc. But step by step.

Instead of blacklisting an instance, it is now banning an instance or an instance has been banned, which sounds much better (in my opinion).

We need to verify whether this is really backward compatible with some benchmarks, currently, I think this is the case since we use for the column families the ordinals. The API methods are only use internally, but it needs to be verified ofc.

> **Note:**  I'm of course also open for another term if you think it is better. This is more a proposal, which I found when searching for [alternatives to blacklisting](https://www.thesaurus.com/browse/blacklist)

## Related issues

related https://github.com/camunda-cloud/runbook/issues/54

BTW also kudos to @korthout who brought this up a while ago.

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: protocol/revapi.json [hunk 1]
```diff
@@ -73,6 +73,18 @@
...
+          "code": "java.method.removed",
+          "old": "io.camunda.zeebe.protocol.record.intent.IncidentIntent::shouldBlacklistInstanceOnError()",
+          "justification": "This method had a racial loaded term. It was replaced with a better name, this shouldn't been an issue as it was for internally use only."
+        },
+        {
+          "ignore": true,
+          "code": "java.method.removed",
+          "old": "io.camunda.zeebe.protocol.record.intent.JobIntent::shouldBlacklistInstanceOnError()",
+          "justification": "This method had a racial loaded term. It was replaced with a better name, this shouldn't been an issue as it was for internally use only."
+        },
```
Comment (remcowesterhoud) [inline protocol/revapi.json:87]: ❓ Why do we need these twice? They are also on line 206-215, except here it references the return type. There's only 1 method in the intents.

**Review by remcowesterhoud — APPROVED** [review]
Love the changes @Zelldon 🚀 

I have some questions and comments. Please have a look at those, nothing blocking.

🔧 I did a text search for `blacklist` and found some occurrences still in the `BannedInstanceStateTest` and the `zeebe.json` Grafana dashboard.

File: engine/src/main/java/io/camunda/zeebe/engine/metrics/BannedInstanceMetrics.java [hunk 2]
```diff
@@ -9,27 +9,27 @@
...
-public final class BlacklistMetrics {
+public final class BannedInstanceMetrics {
 
-  private static final Gauge BLACKLISTED_INSTANCES_COUNTER =
+  private static final Gauge BANNED_INSTANCES_COUNTER =
       Gauge.build()
           .namespace("zeebe")
-          .name("blacklisted_instances_total")
-          .help("Number of blacklisted instances")
+          .name("banned_instances_total")
```
[the remark this answers is not in the document]
Comment (Zelldon) [reply engine/src/main/java/io/camunda/zeebe/engine/metrics/BannedInstanceMetrics.java:17]: I will take care of this in a follow up. We can use a max of both I think.

File: protocol/revapi.json (hunk 1 above)
Comment (Zelldon) [reply protocol/revapi.json:87]: Ups this was a left over sorry.

**Comment by Zelldon** [comment]
I was executing an update test with a banned instances and run into this https://github.com/camunda/zeebe/issues/13101. It is not related to the changes itself, but something we need to fix as well.

**Comment by Zelldon** [comment]
Besides the bug I found https://github.com/camunda/zeebe/pull/13068#issuecomment-1590015478 I think there is no issue with changing this name, since the exception also indicates it founds the correct column family:

```
java.lang.RuntimeException: io.camunda.zeebe.db.ZeebeDbInconsistentException: Key DbLong{4503599627370497} in ColumnFamily BANNED_INSTANCE already exists
```

Would you agree @remcowesterhoud ?

**Comment by remcowesterhoud** [comment]
> Besides the bug I found [#13068 (comment)](https://github.com/camunda/zeebe/pull/13068#issuecomment-1590015478) I think there is no issue with changing this name, since the exception also indicates it founds the correct column family:
> 
> ```
> java.lang.RuntimeException: io.camunda.zeebe.db.ZeebeDbInconsistentException: Key DbLong{4503599627370497} in ColumnFamily BANNED_INSTANCE already exists
> ```
> 
> Would you agree @remcowesterhoud ?

I agree, if there's data in the `ColumnFamily` it means the rename isn't causing issues.

**Comment by Zelldon** [comment]
Just want to share that I discussed this with @megglos yesterday. We agreed on backporting this change also to our older version so we can get rid of the term, also in older version.

**Pull request merged**